--- a/rhino/src/main/java/org/mozilla/javascript/regexp/NativeRegExp.java
+++ b/rhino/src/main/java/org/mozilla/javascript/regexp/NativeRegExp.java
@@ -2460,7 +2460,7 @@
                     char dch = downcase(ch);
                     if (ch != uch) addCharacterToCharSet(charSet, uch);
                     if (ch != dch) addCharacterToCharSet(charSet, dch);
-                    if (++ch == 0) break; // overflow
+                    if (++ch == 0) break; // overflow (RHBZ#1011947)
                 }
             } else addCharacterRangeToCharSet(charSet, start, end);
         }
