--- a/buildSrc/src/main/groovy/rhino.library-conventions.gradle
+++ b/buildSrc/src/main/groovy/rhino.library-conventions.gradle
@@ -29,6 +29,16 @@
             "Implementation-Title":  "Mozilla Rhino",
             "Implementation-Vendor": "Mozilla Foundation",
             "Implementation-URL": "http://github.com/mozilla/rhino",
+            // OSGi metadata from Eclipse Orbit (packaging)
+            "Bundle-Vendor": "%Bundle-Vendor.0",
+            "Bundle-Localization": "plugin",
+            "Bundle-RequiredExecutionEnvironment": "JavaSE-17",
+            "Bundle-Name": "%Bundle-Name.0",
+            "Bundle-SymbolicName": "org.mozilla.javascript",
+            "Bundle-Version": project.version,
+            "Bundle-ManifestVersion": "2",
+            "Export-Package": "org.mozilla.classfile,org.mozilla.javascript,org.mozilla.javascript.annotations,org.mozilla.javascript.ast,org.mozilla.javascript.commonjs.module,org.mozilla.javascript.commonjs.module.provider,org.mozilla.javascript.config,org.mozilla.javascript.debug,org.mozilla.javascript.optimizer,org.mozilla.javascript.regexp,org.mozilla.javascript.serialize,org.mozilla.javascript.typedarrays,org.mozilla.javascript.xml,org.mozilla.javascript.lc.type",
+            "Import-Package": "javax.swing;resolution:=optional,javax.swing.border;resolution:=optional,javax.swing.event;resolution:=optional,javax.swing.filechooser;resolution:=optional,javax.swing.table;resolution:=optional,javax.swing.text;resolution:=optional,javax.swing.tree;resolution:=optional,javax.xml.namespace;resolution:=optional,javax.xml.parsers;resolution:=optional,javax.xml.transform;resolution:=optional,javax.xml.transform.dom;resolution:=optional,javax.xml.transform.stream;resolution:=optional,org.w3c.dom;resolution:=optional,org.xml.sax;resolution:=optional",
         )
     }
 }
