--- a/examples/src/main/java/Matrix.java
+++ b/examples/src/main/java/Matrix.java
@@ -214,7 +214,7 @@
      * <p>Use the convenience method from Context that takes care of calling toString, etc.
      */
     @Override
-    public Object getDefaultValue(Class<?> typeHint) {
+    public Object getDefaultValue(Class typeHint) {
         return "[object Matrix]";
     }
 
--- a/rhino/src/main/java/org/mozilla/javascript/WrapFactory.java
+++ b/rhino/src/main/java/org/mozilla/javascript/WrapFactory.java
@@ -46,7 +46,7 @@
      *     class, staticType will be used instead.
      * @return the wrapped value.
      */
-    public Object wrap(Context cx, Scriptable scope, Object obj, Class<?> staticType) {
+    public Object wrap(Context cx, Scriptable scope, Object obj, Class staticType) {
         return wrap(cx, scope, obj, TypeInfoFactory.GLOBAL.create(staticType));
     }
 
--- a/rhino-tools/src/main/java/org/mozilla/javascript/tools/shell/JavaPolicySecurity.java
+++ b/rhino-tools/src/main/java/org/mozilla/javascript/tools/shell/JavaPolicySecurity.java
@@ -46,7 +46,7 @@
         }
 
         @Override
-        public void linkClass(Class<?> cl) {
+        public void linkClass(Class cl) {
             resolveClass(cl);
         }
     }
--- a/rhino-xml/src/main/java/org/mozilla/javascript/xmlimpl/Namespace.java
+++ b/rhino-xml/src/main/java/org/mozilla/javascript/xmlimpl/Namespace.java
@@ -86,7 +86,7 @@
     }
 
     @Override
-    public Object getDefaultValue(Class<?> hint) {
+    public Object getDefaultValue(Class hint) {
         return uri();
     }
 
--- a/rhino-xml/src/main/java/org/mozilla/javascript/xmlimpl/QName.java
+++ b/rhino-xml/src/main/java/org/mozilla/javascript/xmlimpl/QName.java
@@ -119,7 +119,7 @@
     }
 
     @Override
-    public Object getDefaultValue(Class<?> hint) {
+    public Object getDefaultValue(Class hint) {
         return toString();
     }
 
--- a/rhino-xml/src/main/java/org/mozilla/javascript/xmlimpl/XMLObjectImpl.java
+++ b/rhino-xml/src/main/java/org/mozilla/javascript/xmlimpl/XMLObjectImpl.java
@@ -110,7 +110,7 @@
     }
 
     @Override
-    public final Object getDefaultValue(Class<?> hint) {
+    public final Object getDefaultValue(Class hint) {
         return this.toString();
     }
 
